### PR TITLE
Define model as a property of all estimators

### LINKED ISSUE #371: Add get_model method to Classifer
**Is your feature request related to a problem? Please describe.**
Adaptive attacks often require modification of the underlying model, such as by removing the softmax output or replacing operations in the network (e.g. replacing log(exp(x)) with x). However, it is difficult to access the underlying model after wrapping it in Classifier.

**Describe the solution you'd like**
A `get_model()` method in Classifier that returns the original underlying model.

**Describe alternatives you've considered**
I have used `classifier._model` to get access to the model and modify it for TF models.  However, the PyTorch wrapper adds an additional layer and `classifier._model._model` is needed.

### PR DESCRIPTION
**Pull request by beat-buesser**
Signed-off-by: Beat Buesser <beat.buesser@ie.ibm.com>

# Description

This pull request define model as a property of all estimators. This enables access to the model in a classifier.

Fixes #371 

## Type of change

Please check all relevant options.

- [ ] Bug fix (non-breaking)
- [x] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by beat-buesser** [comment]
@seanpmorgan Thank you very much for your review and suggestions!
